odl/solvers/functional/functional.py

```diff
@@ -500,7 +501,7 @@ def __init__(self, func, scalar):
             raise TypeError('`func` {!r} is not a `Functional` instance'
                             ''.format(func))
 
-        scalar = func.range.element(scalar)
+        scalar = func.domain.field.element(scalar)
```

Comment:
Currently yes, but once we got the math straight, we might in fact want allow spaces over the complex numbers as `domain`, and in that case, we want to cast to the field of the domain, not the range (which we want to keep as "always real", I suppose).